### PR TITLE
refactor(grid,paging): internal paginator API use

### PR DESCRIPTION
**Pull request by damyanpetev**
Isolated changes @kdinev made in https://github.com/IgniteUI/igniteui-angular/pull/12229

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/grid-base.directive.ts [hunk 1]
```diff
@@ -1859,8 +1859,6 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     }
 
     /**
-     * @deprecated in version 12.1.0. Use `page` property form `paginator` component instead
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/grid-base.directive.ts:1862]: This should be marked internal.

File: projects/igniteui-angular/src/lib/grids/grid-base.directive.ts [hunk 2]
```diff
@@ -1885,8 +1883,6 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     }
 
     /**
-     * @deprecated in version 12.1.0. Use `perPage` property from `paginator` component instead
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/grid-base.directive.ts:1888]: This should be marked internal

File: projects/igniteui-angular/src/lib/grids/grid/grid-add-row.spec.ts [hunk 3]
```diff
@@ -102,7 +102,7 @@ describe('IgxGrid - Row Adding #grid', () => {
             grid.notifyChanges(true);
             fixture.detectChanges();
 
-            grid.paginator.perPage = 7;
+            grid.perPage = 7;
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/grid/grid-add-row.spec.ts:105]: The spec changes can be reverted.

File: projects/igniteui-angular/src/lib/grids/grid/grid.master-detail.spec.ts [hunk 4]
```diff
@@ -886,10 +886,10 @@ describe('IgxGrid Master Detail #grid', () => {
                 ];
                 fix.componentInstance.paging = true;
                 fix.detectChanges();
-                grid.paginator.perPage = 5;
+                grid.perPage = 5;
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/grid/grid.master-detail.spec.ts:889]: Again, spec changes can be reverted.

**Review by kdinev — CHANGES_REQUESTED** [review]

**Review by kdinev — APPROVED** [review]

**Comment by kdinev** [comment]
@damyanpetev Ready to be merged?

File: projects/igniteui-angular/src/lib/grids/grid-base.directive.ts (hunk 1 above)
Comment (damyanpetev) [reply projects/igniteui-angular/src/lib/grids/grid-base.directive.ts:1862]: We ended up keeping it deprecated as-is, right? So those two comments are now outdated, right?

Comment (hanastasov) [reply projects/igniteui-angular/src/lib/grids/grid-base.directive.ts:1877]: If this is so (keeping the deprecated props/methods as-is), is this valid for the outputs too? Like still emitting `perPageChange` grid event. My expectation was those will be removed.

File: projects/igniteui-angular/src/lib/grids/grid-base.directive.ts [hunk 5]
```diff
@@ -4702,18 +4714,20 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     }
 
     /**
-     * @deprecated in version 12.1.0. Use the corresponding method exposed by the `igx-paginator`
+     * @deprecated in version 12.1.0. Use the corresponding method `paginate()` exposed by the `igx-paginator`
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/grid-base.directive.ts:4717]:
```suggestion
     * @deprecated in version 12.1.0. Use the corresponding method `paginate()` exposed by the `igx-paginator` instance.
```

**Review by kdinev — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/grids/grid-base.directive.ts (hunk 1 above)
Comment (hanastasov) [reply projects/igniteui-angular/src/lib/grids/grid-base.directive.ts:1877]: Those are not used internally.

Comment (zdrawku) [reply projects/igniteui-angular/src/lib/grids/grid-base.directive.ts:1877]: After discussing this with @ddincheva and @damyanpetev we are going to merge this PR as is, and do separate PR changes for the deprecations. (This is only API refactoring)

**Review by zdrawku — APPROVED** [review]

**Comment by zdrawku** [comment]
I should've squash merge it, apologies

**Pull request merged**